### PR TITLE
Fix links in javadoc and documentations

### PR DESCRIPTION
**Pull request by marcwrobel**
- fix broken links,
- use direct links instead of redirects,
- use HTTPS where possible.

### CODE REVIEW

File: checkstyle.xml [hunk 1]
```diff
@@ -26,25 +26,25 @@
   <property name="localeLanguage" value="en" />
 
   <!-- Checks that a package-info.java file exists for each package. -->
-  <!-- See http://checkstyle.sourceforge.net/config_javadoc.html -->
+  <!-- See https://checkstyle.org/config_javadoc.html -->
```
Comment (garydgregory) [inline checkstyle.xml:29]: Do you mean https://checkstyle.org/config_javadoc.html#JavadocPackage ?

File: checkstyle.xml [hunk 2]
```diff
@@ -26,25 +26,25 @@
...
   <!-- Checks for Tab characters -->
-  <!-- See http://checkstyle.sourceforge.net/config_whitespace.html#FileTabCharacter -->
+  <!-- See https://checkstyle.org/config_whitespace.html#FileTabCharacter -->
   <module name="FileTabCharacter">
     <property name="fileExtensions" value="java" />
   </module>
 
   <!-- Checks for white space at the end of the line -->
-  <!-- See http://checkstyle.sourceforge.net/config_regexp.html -->
+  <!-- See https://checkstyle.org/config_regexp.html -->
```
Comment (garydgregory) [inline checkstyle.xml:47]: Probably https://checkstyle.org/config_regexp.html#RegexpSingleline ?

File: checkstyle.xml [hunk 3]
```diff
@@ -68,7 +68,7 @@
   </module>
 
   <!-- Checks for Size Violations. -->
-  <!-- See http://checkstyle.sf.net/config_sizes.html -->
+  <!-- See https://checkstyle.org/config_sizes.html -->
```
Comment (garydgregory) [inline checkstyle.xml:71]: https://checkstyle.org/config_sizes.html#LineLength

File: checkstyle.xml [hunk 4]
```diff
@@ -81,13 +81,13 @@
   <module name="TreeWalker">
 
     <!-- Checks for Javadoc comments. -->
-    <!-- See http://checkstyle.sf.net/config_javadoc.html -->
+    <!-- See https://checkstyle.org/config_javadoc.html -->
```
Comment (garydgregory) [inline checkstyle.xml:84]: https://checkstyle.org/config_javadoc.html#JavadocMethod

File: checkstyle.xml [hunk 5]
```diff
@@ -100,7 +100,7 @@
     </module>
 
     <!-- Checks for Naming Conventions. -->
-    <!-- See http://checkstyle.sf.net/config_naming.html -->
+    <!-- See https://checkstyle.org/config_naming.html -->
```
Comment (garydgregory) [inline checkstyle.xml:103]: https://checkstyle.org/config_naming.html#ConstantName ?

File: src/site/xdoc/index.xml [hunk 6]
```diff
@@ -148,9 +148,9 @@
             methods for provider interfaces (like <code>FileObject</code>). If you implement a VFS provider and use the
             corresponding <code>Abstract*</code> or <code>Default*</code> classes, there should be no need to modify
             the code or recompile the provider. The TarFileProvider is one known exception to compatibility with 2.0.
-            See the <a href="https://dist.apache.org/repos/dist/dev/commons/vfs/RELEASE-NOTES.txt">Release Notes</a> and the
-            <a href="commons-vfs2/clirr-report.html">Clirr Report</a> for details. VFS 2.1 adds a new read-only provider
-            for the Apache Hadoop (HDFS) File system.
+            See the <a href="https://gitbox.apache.org/repos/asf?p=commons-vfs.git;a=blob_plain;f=RELEASE-NOTES.txt;hb=HEAD">
```
Comment (garydgregory) [inline src/site/xdoc/index.xml:151]: -1: Please Use https://dist.apache.org/repos/dist/release/commons/vfs/RELEASE-NOTES.txt

**Review by garydgregory — CHANGES_REQUESTED** [review]
Hi @marcwrobel 
Thank you for your PR. One important fix to do (see "-1") and other comments.

File: checkstyle.xml (hunk 1 above)
Comment (marcwrobel) [reply checkstyle.xml:29]: Seems better, changed.

File: checkstyle.xml (hunk 2 above)
Comment (marcwrobel) [reply checkstyle.xml:47]: Seems better, comment updated.

File: checkstyle.xml (hunk 3 above)
Comment (marcwrobel) [reply checkstyle.xml:71]: Seems better, comment updated.

File: checkstyle.xml (hunk 4 above)
Comment (marcwrobel) [reply checkstyle.xml:84]: No, this comment is also for MissingJavadocMethod.

File: checkstyle.xml (hunk 5 above)
Comment (marcwrobel) [reply checkstyle.xml:103]: No, this comment is also for LocalFinalVariableName, LocalVariableName...

File: src/site/xdoc/index.xml (hunk 6 above)
Comment (marcwrobel) [reply src/site/xdoc/index.xml:151]: Done.

**Pull request merged**